File: content/04/code/js/03-first-scenario-rewritten/features/step_definitions/steps.js
const { Given, When, Then } = require('@cucumber/cucumber')
const { assertThat, is } = require('hamjest')

const { Person, Network } = require('../../src/shouty')

Given('Lucy is {int} metres from Sean', function (distance) {
  this.network = new Network()
  this.lucy    = new Person(this.network)

  this.lucy.moveTo(distance)
})

When('Sean shouts {string}', function (message) {
  sean = new Person(this.network)
  sean.shout(message)
  this.messageFromSean = message
})

Then('Lucy should hear Sean\'s message', function () {
  assertThat(this.lucy.messagesHeard(), is([this.messageFromSean]))
})

File: content/bdd-with-cucumber/04/code/ruby/01-description-added/features/step_definitions/steps.rb
require 'shouty'

Given("Lucy is {int} metres from Sean") do |distance|
  @network = Shouty::Network.new
  @lucy    = Shouty::Person.new(@network)
  @sean    = Shouty::Person.new(@network)
  
  @lucy.move_to(distance)
end

When("Sean shouts {string}") do |message|
  @sean.shout(message)
  @message_from_sean = message
end

Then("Lucy should hear Sean's message") do
  expect(@lucy.messages_heard).to eq [@message_from_sean]
end
